File: erettsegi/kozep/2022_may/java/Robot.java
import module java.base;

public class Robot {

    public static void main(String[] args) {
        var parancsok = System.console().readLine("Kérem a robot parancsait: ");
        var betuStatok = new HashMap<Character, Integer>();

        for(var i = 0; i < parancsok.length(); ++i) {
            var parancsBetu = parancsok.charAt(i);

            betuStatok.put(parancsBetu, betuStatok.getOrDefault(parancsBetu, 0) + 1);
        }

        for(var entry : betuStatok.entrySet()) {
            System.out.println(entry.getKey() + " betűk száma: " + entry.getValue());
        }

        var vegsoX = 0;
        var vegsoY = 0;
        for(var i = 0; i < parancsok.length(); ++i) {
            var parancs = parancsok.charAt(i);
            vegsoX = parancs == 'N' ? vegsoX - 1 : parancs == 'K' ? vegsoX + 1 : vegsoX;
            vegsoY = parancs == 'E' ? vegsoY + 1 : parancs == 'D' ? vegsoY - 1 : vegsoY;
        }

        var xParancsBetu = vegsoX > 0 ? "K" : "N";
        var egyszerusitettXParancsok = xParancsBetu.repeat(Math.abs(vegsoX));
        var yParancsBetu = vegsoY > 0 ? "E" : "D";
        var egyszerusitettYParancsok = yParancsBetu.repeat(Math.abs(vegsoY));

        System.out.println("Egy legrövidebb út parancsszava: " + egyszerusitettXParancsok + egyszerusitettYParancsok);
    }
}

File: erettsegi/kozep/2022_may/python/robot.py
from collections import Counter

parancsok = input('Kérem a robot parancsait: ')
betu_statok = Counter(parancsok)

for betu, db in betu_statok.items():
    print(f'{betu} betűk száma: {db}')

vegso_x = 0
vegso_y = 0
for parancs in parancsok:
    vegso_x = vegso_x - 1 if parancs == 'N' else vegso_x + 1 if parancs == 'K' else vegso_x
    vegso_y = vegso_y + 1 if parancs == 'E' else vegso_y - 1 if parancs == 'D' else vegso_y

x_parancs_betu = 'K' if vegso_x > 0 else 'N'
egyszerusitett_x_parancsok = x_parancs_betu * abs(vegso_x)
y_parancs_betu = 'E' if vegso_y > 0 else 'D'
egyszerusitett_y_parancsok = y_parancs_betu * abs(vegso_y)

print(f'Egy legrövidebb út parancsszava: {egyszerusitett_x_parancsok}{egyszerusitett_y_parancsok}')
